### PR TITLE
feat: improve constructor documentation

### PR DESCRIPTION
**Pull request by nomagick**
Allow use_with params for `DocumentArrayMemmap`:
* key_length:  DocumentArrayMemmap(path, key_length=`key_length`)

Documents how to change the storage path.

### CODE REVIEW

File: executor.py [hunk 1]
```diff
@@ -18,17 +18,21 @@ class SimpleIndexer(Executor):
     def __init__(
         self,
         match_args: Optional[Dict] = None,
+        key_length: int = 64,
+        override_storage_path: str = None,
```
Comment (winstonww) [inline executor.py:22]: Might be better to add `Optional` typing to `override_storage_path: Optional[str] = None`

**Review by winstonww — COMMENTED** [review]
LGTM, just a minor comment.

File: executor.py (hunk 1 above)
Comment (nomagick) [reply executor.py]: Cool, new to modern type annotated python though

**Review by winstonww — DISMISSED** [review]
LGTM!

File: executor.py [hunk 2]
```diff
@@ -18,17 +18,21 @@ class SimpleIndexer(Executor):
...
         :param match_args: the arguments to `DocumentArray`'s match function
+        :param override_storage_path: the `path` argument to `DocumentArrayMemmap`'s constructor
+        :param key_length: the `key_length` keyword argument to `DocumentArrayMemmap`'s constructor
         """
         super().__init__(**kwargs)
 
         self._match_args = match_args or {}
         self._storage = DocumentArrayMemmap(
-            self.workspace, key_length=kwargs.get('key_length', 64)
+            override_storage_path or self.workspace, key_length = key_length
```
Comment (winstonww) [inline executor.py:35]: On second thought, if we set the` workspace` path when initializing `SimpleIndexer` as follows:
```python
Flow.add(uses=`jinahub://SimpleIndexer', uses_metas={'workspace': workspace})
```
then we don't need `override_storage_path`?

Comment (nomagick) [reply executor.py:35]: Yes, technically.  
But using workspace as storage root is an implementation detail to users. I did not know how to override this path before looking into the source code. 
And as a newbie, I also did not know that workspace could be overridden using `uses_metas`...
So I think having a `uses_with` parameter for path overriding is a reasonable approach and is more intuitive to users (who don't know all the internals).

Comment (hanxiao) [reply executor.py:35]: reject, this should be a documentation job

**Review by hanxiao — DISMISSED** [review]
reject on the overriding path

**Review by winstonww — APPROVED** [review]
We are having some issues with `uses_metas` at the moment:
https://github.com/jina-ai/jina/issues/3857

Another way to do this is to:
```python
Flow().add(uses='jinahub://SimpleIndexer', workspace=workspace)
```
I will approve this PR, but feel free to update the description to include the above.

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**